src/twisted/words/protocols/irc.py

```diff
@@ -405,6 +405,8 @@ def dataReceived(self, data):
         says CRLF.  (Also, the flexibility of LineReceiver to turn "line mode"
         on and off was not required.)
         """
+        if isinstance(data, bytes):
+            data = data.decode("utf-8")
```

Comment:
Naively decoding input as UTF-8 sounds like a bad idea without ability to set user-defined (probably per-channel) fallback encoding. Normal IRC clients tend to work so that they first try to decode as UTF-8 and if that fails, fallback to configurable legacy 8-bit encoding